easybuild/tools/toolchain/mpi.py

```diff
@@ -191,7 +191,8 @@ def mpi_cmd_for(self, cmd, nr_ranks):
             tmpdir = tempfile.mkdtemp(prefix='eb-mpi_cmd_for-')
 
             # set temporary dir for mdp
-            env.setvar('I_MPI_MPD_TMPDIR', tmpdir)
+            # note: this needs to be kept *short*, to avoid mpirun failing with "socket.error: AF_UNIX path too long"
+            env.setvar('I_MPI_MPD_TMPDIR', tempfile.gettempdir())
```

Comment:
just checking, using `tempfile.gettmpdir()` rather than `tmpdir` fixes the problem, i.e. a value like `/tmp/easybuild-F_B4M3` *is* short enough